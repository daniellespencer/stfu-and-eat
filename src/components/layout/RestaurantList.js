import React, { Fragment } from 'react';
import { Jumbotron, Container } from 'react-bootstrap';
import RecommendationItem from './RecommendationItem';

const RestaurantList = () => {
    return (
        <Fragment>
            <Jumbotron fluid className='listBanner' >
            <Container >
                <h1>Our List</h1>
               
            </Container>
        </Jumbotron>

        <RecommendationItem />
        
        </Fragment>
    )
}

export default RestaurantList
