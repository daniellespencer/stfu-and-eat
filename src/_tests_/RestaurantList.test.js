import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';

import RestaurantList from '../components/layout/RestaurantList';

afterEach(cleanup);

it("renders correctly", () => {
    shallow(<RestaurantList />);
})

describe("Input value", () => {
    it("updates on change", () => {
        const { queryByPlaceholderText } = render(<RestaurantList />)

        const searchInput = queryByPlaceholderText('Search our list of restaurants ...');

        fireEvent.change(searchInput, {target: {value: "test"}})

        expect(searchInput.value).toBe("test")
    })
})