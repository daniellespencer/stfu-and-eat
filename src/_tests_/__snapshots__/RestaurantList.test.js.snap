// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantList renders multiple restaurants 1`] = `
Array [
  <div
    className="listBanner jumbotron jumbotron-fluid"
  >
    <div
      className="container"
    >
      <h1>
        Our List
      </h1>
    </div>
  </div>,
  <div
    className="container"
  >
    <input
      className="form-control"
      id="search-input"
      onChange={[Function]}
      placeholder="Search our list of restaurants ..."
      type="text"
    />
    <div
      className="justify-content-sm-center row row-cols-md-3 row-cols-1"
    >
      <div>
        <div
          className="shadow mb-5 mt-5  card"
        >
          <img
            className="card-img-top img-fluid"
            src="rec-card-default.jpg"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Benton Park Cafe
            </div>
            <p
              className="card-text"
            >
              Neighborhood: 
              Benton Park
               
              <br />
              Cuisine: 
              American
            </p>
            <button
              className="btn-success btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Restaurant website
            </button>
          </div>
        </div>
      </div>
      <div>
        <div
          className="shadow mb-5 mt-5  card"
        >
          <img
            className="card-img-top img-fluid"
            src="rec-card-default.jpg"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Kalbi Taco Shack
            </div>
            <p
              className="card-text"
            >
              Neighborhood: 
              Benton Park
               
              <br />
              Cuisine: 
              Asian Fusion
            </p>
            <button
              className="btn-success btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Restaurant website
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`RestaurantList renders no restaurants 1`] = `
Array [
  <div
    className="listBanner jumbotron jumbotron-fluid"
  >
    <div
      className="container"
    >
      <h1>
        Our List
      </h1>
    </div>
  </div>,
  <div
    className="container"
  >
    <input
      className="form-control"
      id="search-input"
      onChange={[Function]}
      placeholder="Search our list of restaurants ..."
      type="text"
    />
    <div
      className="justify-content-sm-center row row-cols-md-3 row-cols-1"
    />
  </div>,
]
`;
